### DISCUSSION TITLE
Failed to flash: Cannot find a Crazyradio Dongle

### DISCUSSION THREAD

**Discussion by Amin-Yazdanshenas** [bitcraze/discussions#1067] [Q&A]
Hello,

I am encountering an issue when trying to flash the firmware to a Crazyflie using the `make cload` command. Below is the command I used and the error message I received:

Command: 
```
crazyflie-firmware$ CLOAD_CMDS="-w radio://0/80/2M/E7E7E7E7E7" make cload
``` 

Error:
``` 
tools/kbuild/Makefile.kbuild:147: warning: overriding recipe for target 'cload'
Makefile:136: warning: ignoring old recipe for target 'cload'
make[1]: Entering directory '/home/aminys/crazyflie/crazyflie-firmware/build'
python.exe -m cfloader -w radio://0/80/2M/E7E7E7E703 flash  cf2.bin stm32-fw
Reset to bootloader mode ...
Failed to flash: Cannot find a Crazyradio Dongle
make[1]: *** [/home/aminys/crazyflie/crazyflie-firmware/Makefile:136: cload] Error 255
make[1]: Leaving directory '/home/aminys/crazyflie/crazyflie-firmware/build'
make: *** [tools/kbuild/Makefile.kbuild:150: sub-make] Error 2
``` 
My setup:
- Crazyradio version: 2.0
- Operating System: WSL2/Ubuntu 20.04

I have verified that my Crazyradio 2.0 dongle is correctly plugged in and the drivers are up to date. The issue persists despite multiple attempts and ensuring the URI matches my Crazyflie's configuration. Any assistance in resolving this issue would be greatly appreciated.

**Comment by Amin-Yazdanshenas** [comment]
@ataffanel 
ataffanel

**Comment by knmcguire** [comment]
Hi, so you are using WSL on Windows right? That doesn't recognize the usb device that you plug into windows (check that with `lsusb`), as it says it can't see the crazyradio.

 So there are two ways to approach this:
- Using python on the windows side. You can use `python.exe -m cfloader flash build/cf2.bin stm32-fw -w radio://0/80/2M/E7E7E7E7E7 (you need to install the cfclient on windows with pip3 install cfclient). You do need to install the [crazyradio drivers with zadig](https://www.bitcraze.io/documentation/repository/crazyradio-firmware/master/building/usbwindows/) on windows.
- Use usbipd to connect the usb to your wsl: https://github.com/dorssel/usbipd-win/wiki/WSL-support (don't forget about [udev rules](https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/installation/usb_permissions/) then)
